lib/cipherstash/protect/analysis/token_filters.rb

```diff
@@ -22,6 +22,17 @@ def perform(str_or_array)
                 (token.length - token_length + 1).times do |i|
                   out << token[i, token_length]
                 end
+
+                split_token = token.split("")
+                init_ngram = [split_token[0, token_length].join]
+                rest = split_token[token_length, split_token.length]
```

Comment:
In Ruby, `-1` as an array index means "the end of the array".

So you can do `rest = split_token[token_length, -1]` which I think is more idiomatic Ruby.